### PR TITLE
fix: Integration Bali PP

### PR DESCRIPTION
**Pull request by joanestebanr**
## Description

- Issues a long integration: 
- Wrong process of unknown error field `PError` on certificate header
- Fix use of aggsender log (to be able to filter by module)
- Fix a bad condition to show a warning of a reopen certificate
- Improved test for epoch notifier
- Fix certificate on local DB but AggLayer have none certificate (CDK-604)
- Add logs to `synv/evmdownloader.go` to have more information on error: `error calling FilterLogs to eth client: invalid params`

### CODE REVIEW

File: agglayer/proof_generation_error.go [hunk 1]
```diff
@@ -38,6 +38,15 @@ type PPError interface {
 	String() string
 }
 
+type GenericError struct {
+	Key    string
+	Values string
```
Comment (ToniRamirezM) [inline agglayer/proof_generation_error.go]: Value or Values?

File: agglayer/types.go [hunk 2]
```diff
@@ -654,7 +654,9 @@ func (c *CertificateHeader) UnmarshalJSON(data []byte) error {
 
 				ppError = p
 			default:
-				return fmt.Errorf("invalid error type: %s", key)
+				valueStr, _ := json.Marshal(value)
+
+				ppError = &GenericError{Key: key, Values: string(valueStr)}
```
Comment (ToniRamirezM) [inline agglayer/types.go]: Here you mix Value and Values

File: aggsender/aggsender.go [hunk 3]
```diff
@@ -202,7 +202,6 @@ func (a *AggSender) sendCertificate(ctx context.Context) (*agglayer.SignedCertif
 
 	a.saveCertificateToFile(signedCertificate)
 	a.log.Infof("certificate ready to be send to AggLayer: %s", signedCertificate.String())
```
Comment (ToniRamirezM) [inline aggsender/aggsender.go:204]: You should decide if logs start with capital letters or not

File: aggsender/aggsender.go [hunk 4]
```diff
@@ -638,6 +637,11 @@ func (a *AggSender) checkLastCertificateFromAgglayer(ctx context.Context) error
...
 		return nil
 	}
+	// CASE 2.1: certificate in storage but not in agglayer
+	// this is a non-sense, so thrown an error
+	if localLastCert != nil && aggLayerLastCert == nil {
+		return fmt.Errorf("recovery: certificate in storage but not in agglayer. Inconsistency")
+	}
 	// CASE 3: aggsender stopped between sending to agglayer and storing on DB
 	if aggLayerLastCert.Height == localLastCert.Height+1 {
 		a.log.Infof("recovery: AggLayer have next cert (height:%d), so is a recovery case: storing cert: %s",
```
Comment (ToniRamirezM) [inline aggsender/aggsender.go:647]:
```suggestion
		a.log.Infof("recovery: AggLayer has next cert (height:%d), so it is a recovery case: storing cert: %s",
```

**Review by ToniRamirezM — APPROVED** [review]

File: agglayer/types.go [hunk 5]
```diff
@@ -654,7 +663,9 @@ func (c *CertificateHeader) UnmarshalJSON(data []byte) error {
 
 				ppError = p
 			default:
-				return fmt.Errorf("invalid error type: %s", key)
+				valueStr, _ := json.Marshal(value)
```
Comment (Stefan-Ethernal) [inline agglayer/types.go:666]: Let's handle this error, just in case. WDYT?

File: aggsender/aggsender.go [hunk 6]
```diff
@@ -584,7 +583,7 @@ func (a *AggSender) updateCertificateStatus(ctx context.Context,
 		localCert.String())
 
 	// That is a strange situation
-	if agglayerCert.Status.IsOpen() == localCert.Status.IsClosed() {
+	if agglayerCert.Status.IsOpen() && localCert.Status.IsClosed() {
 		a.log.Warnf("certificate %s is reopen! from [%s] to [%s]",
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender.go:587]:
```suggestion
		a.log.Warnf("certificate %s is reopened on the agglayer! from [%s] to [%s]",
```

File: agglayer/proof_generation_error.go (hunk 1 above)
Comment (joanestebanr) [reply agglayer/proof_generation_error.go]: Value

File: agglayer/types.go (hunk 2 above)
Comment (joanestebanr) [reply agglayer/types.go]: I have rename to `Value`

File: aggsender/aggsender.go (hunk 3 above)
Comment (joanestebanr) [reply aggsender/aggsender.go:204]: Humm... I think that almost all of them are in lower

**Review by Stefan-Ethernal — APPROVED** [review]

**Pull request merged**